pkg/account/component.go

```diff
@@ -541,3 +543,44 @@ func (c *component) executeActions(ctx context.Context, actions []string) error
 
 	return err
 }
+
+func (c *component) CancelEmailChange(ctx context.Context) error {
+	return c.removeStringAttributeFromUser(ctx, constants.AttrbEmailToValidate)
+}
+
+func (c *component) CancelPhoneNumberChange(ctx context.Context) error {
+	return c.removeStringAttributeFromUser(ctx, constants.AttrbPhoneNumberToValidate)
+}
+
+func (c *component) removeStringAttributeFromUser(ctx context.Context, attr kc.AttributeKey) error {
+	accessToken := ctx.Value(cs.CtContextAccessToken).(string)
+	realm := ctx.Value(cs.CtContextRealm).(string)
+
+	// Get the user representation from Keycloak
+	userKc, err := c.keycloakAccountClient.GetAccount(accessToken, realm)
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Can't get account", "err", err.Error())
+		return err
+	}
+	keycloakb.ConvertLegacyAttribute(&userKc)
+
+	if userKc.GetAttributeString(attr) == nil {
```

Comment:
Did you tested with user profile activated on the realm? I wonder if "missing" attribute is nil or empty with user profile.

Reply:
From my understanding, the function `GetString` from `model_toolbox.go` (in keycloak-client) that is called by `GetAttributeString` already transforms an empty string into a nil, so I don't think I need to treat the empty string case here.

Here is the function as a reference:
```
// GetString gets the first value of a given attribute
func (a Attributes) GetString(key AttributeKey) *string {
	var attrbs = a[key]
	if len(attrbs) == 1 && attrbs[0] == "" {
		// User profile sometimes sets a default empty value
		return nil
	}
	if len(attrbs) > 0 {
		return &attrbs[0]
	}
	return nil
}
```